siliconcompiler/tools/openroad/openroad.py

```diff
@@ -398,38 +395,39 @@ def get_metric_sources(metric):
 
 
 ######
-def get_pex_corners(chip):
-
+def get_library_timing_keypaths(chip, lib):
     step = chip.get('arg', 'step')
     index = chip.get('arg', 'index')
 
-    corners = set()
+    delaymodel = chip.get('asic', 'delaymodel', step=step, index=index)
+    keypaths = {}
     for constraint in chip.getkeys('constraint', 'timing'):
-        pexcorner = chip.get('constraint', 'timing', constraint, 'pexcorner',
-                             step=step, index=index)
-        if pexcorner:
-            corners.add(pexcorner)
+        corners = chip.get('constraint', 'timing', constraint, 'libcorner', step=step, index=index)
+        for corner in corners:
+            if chip.valid('library', lib, 'output', corner, delaymodel):
+                keypaths[constraint] = ('library', lib, 'output', corner, delaymodel)
 
-    return list(corners)
+        if constraint not in keypaths:
+            keypaths[constraint] = ('library', lib, 'output', corners[0], delaymodel)
```

Comment:
Didn't we already see that corners[0] leads to an invalid keypath in the for loop above?
This sets the `keypath[constraint]` to an invalid path as far as I can tell which I'm not sure is intended...

Reply:
this is to ensure a path is marked required so execution will fail on check_manifest.